### PR TITLE
Avoid checking against types in is_dataframe_like

### PR DESCRIPTION
**Pull request by mrocklin**
- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

File: dask/dataframe/utils.py [hunk 1]
```diff
@@ -497,18 +497,18 @@ def _nonempty_series(s, idx=None):
 
 def is_dataframe_like(df):
     """ Looks like a Pandas DataFrame """
-    return set(dir(df)) > {'dtypes', 'columns', 'groupby', 'head'}
+    return set(dir(df)) > {'dtypes', 'columns', 'groupby', 'head'} and not isinstance(df, type)
```
Comment (mrocklin) [inline dask/dataframe/utils.py]: Anyone have a cleaner suggestion for this?

Comment (gsakkis) [reply dask/dataframe/utils.py]: How about
```
import inspect 

def _is_like(obj, includes, excludes=()):
    return (not inspect.isclass(obj)
            and all(hasattr(obj, attr) for attr in includes)
            and not any(hasattr(obj, attr) for attr in excludes))


def is_dataframe_like(df):
    """ Looks like a Pandas DataFrame """
    return _is_like(df, ('dtypes', 'columns', 'groupby', 'head'))


def is_series_like(s):
    """ Looks like a Pandas Series """
    return _is_like(s, ('name', 'dtype', 'groupby', 'head'))


def is_index_like(s):
    """ Looks like a Pandas Index """
    return _is_like(s, ('name', 'dtype'), ('head',))
```

Comment (jcrist) [reply dask/dataframe/utils.py]: You could check for `not hasattr(df, 'mro')` to exclude types, since all classes have an `mro` method.

**Comment by jcrist** [comment]
> do you have any experience with this failure?

No, never seen that before.

File: dask/dataframe/utils.py (hunk 1 above)
Comment (mrocklin) [reply dask/dataframe/utils.py]: I think I'd like to pass on the more generic version.  There are only three things here and I think it's probably simpler to keep things flat.

The `hasattr(df, 'mro')` check seems equivalent to the isinstance call.

Mostly I was concerned about the case where an instantiated dataframe was also a type, but I think I'll just pass on this case.

**Comment by mrocklin** [comment]
Merging this in.  The windows failure is reported in another PR.

**Pull request merged**